docs/index.bs

```diff
@@ -2962,26 +2963,30 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
       1. Let |fetchInstance| be the instance of the [=/fetch=] algorithm representing the ongoing fetch.
+      1. Let |requestBody| be |request|'s [=request/body=].
       1. Assert: |request|'s [=request/destination=] is not "<code>serviceworker</code>".
       1. If |request|'s [=request/destination=] is either "<code>embed</code>" or "<code>object</code>", then:
-          1. Return null.
+          1. Return null and |requestBody|.
       1. Else if |request| is a <a>non-subresource request</a>, then:
 
           Note: If the non-subresource request is under the scope of a service worker registration, application cache is completely bypassed regardless of whether the non-subresource request uses the service worker registration.
 
           1. If |reservedClient| is not null and is an <a>environment settings object</a>, then:
-              1. If |reservedClient| is not a <a>secure context</a>, return null.
+              1. If |reservedClient| is not a <a>secure context</a>, return null and |requestBody|.
           1. Else:
-              1. If |request|'s [=request/url=] is not a <a>potentially trustworthy URL</a>, return null.
-          1. If |request| is a <a>navigation request</a> and the <a lt="navigate">navigation</a> triggering it was initiated with a shift+reload or equivalent, return null.
+              1. If |request|'s [=request/url=] is not a <a>potentially trustworthy URL</a>, return null and |requestBody|.
+          1. If |request| is a <a>navigation request</a> and the <a lt="navigate">navigation</a> triggering it was initiated with a shift+reload or equivalent, return null and |requestBody|.
           1. Set |registration| to the result of running <a>Match Service Worker Registration</a> algorithm passing |request|'s [=request/url=] as the argument.
-          1. If |registration| is null or |registration|'s <a>active worker</a> is null, return null.
+          1. If |registration| is null or |registration|'s <a>active worker</a> is null, return null and |requestBody|.
           1. If |request|'s [=request/destination=] is not {{RequestDestination/"report"}}, set |reservedClient|'s <a>active service worker</a> to |registration|'s <a>active worker</a>.
           1. If |request| is a [=navigation request=], |registration|'s [=navigation preload enabled flag=] is set, |request|'s [=request/method=] is \`<code>GET</code>\`, and |registration|'s [=active worker=]'s <a>set of event types to handle</a> [=set/contains=] <code>fetch</code>, then:
 
               Note: If the above is true except |registration|'s [=active worker=]'s <a>set of event types to handle</a> **does not** contain <code>fetch</code>, then the user agent may wish to show a console warning, as the developer's intent isn't clear.
 
-              1. Let |preloadRequest| be the result of [=request/cloning=] the request |request|.
+              1. Let |preloadRequest| be a copy of |request|, except for its [=request/body=].
```

Comment:
I don't see why we cannot clone here. Body will still have its source afterwards.